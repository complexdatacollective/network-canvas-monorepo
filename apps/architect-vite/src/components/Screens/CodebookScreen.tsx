import { Button } from "@codaco/legacy-ui/components";
import Codebook from "~/components/Codebook/Codebook";
import { Layout } from "~/components/EditorLayout";
import Screen from "~/components/Screen/Screen";
import ControlBar from "../ControlBar";
import CollapsableHeader from "../Screen/CollapsableHeader";

type CodebookScreenProps = {
	layoutId?: string | null;
	onComplete?: () => void;
};

/**
 * This component acts as an index for types. i.e. Nodes and Edges,
 * and links to the EditType.
 */
const CodebookScreen = ({ layoutId = null, onComplete = () => {} }: CodebookScreenProps) => {

	const buttons = [
		<Button key="done" onClick={onComplete} color="platinum">
			Close
		</Button>,
	];

	return (
		<Screen layoutId={layoutId} footer={<ControlBar buttons={buttons} />} onComplete={onComplete}>
			<CollapsableHeader
				collapsedState={
					<div className="stage-heading stage-heading--collapsed stage-heading--shadow">
						<Layout>
							<h2>Codebook</h2>
						</Layout>
					</div>
				}
			>
				<div className="stage-heading stage-heading--inline">
					<Layout>
						<h1 className="screen-heading">Codebook</h1>
						<p>
							Below you can find an overview of the node and edge types that you have defined while creating your
							interview. Entities that are unused may be deleted.
						</p>
					</Layout>
				</div>
			</CollapsableHeader>
			<Layout>
				<Codebook />
			</Layout>
		</Screen>
	);
};


export default CodebookScreen;
