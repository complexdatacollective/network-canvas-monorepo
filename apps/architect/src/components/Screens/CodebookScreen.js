import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@codaco/ui/lib/components';
import Codebook from '@components/Codebook/Codebook';
import Screen from '@components/Screen/Screen';
import { Layout } from '@components/EditorLayout';
import ControlBar from '../ControlBar';
import CollapsableHeader from '../Screen/CollapsableHeader';

/**
 * This component acts as an index for types. i.e. Nodes and Edges,
 * and links to the EditType.
 */
const CodebookScreen = (props) => {
  const {
    layoutId,
    onComplete,
  } = props;

  const buttons = [
    <Button
      key="done"
      onClick={onComplete}
      color="platinum"
    >
      Close
    </Button>,
  ];

  return (
    <Screen
      layoutId={layoutId}
      footer={<ControlBar buttons={buttons} />}
      onComplete={onComplete}
    >
      <CollapsableHeader
        collapsedState={(
          <div className="stage-heading stage-heading--collapsed stage-heading--shadow">
            <Layout>
              <h2>Codebook</h2>
            </Layout>
          </div>
        )}
      >
        <div className="stage-heading stage-heading--inline">
          <Layout>
            <h1 className="screen-heading">Codebook</h1>
            <p>
              Below you can find an overview of the node and edge types that you have
              defined while creating your interview. Entities that are unused may be deleted.
            </p>
          </Layout>
        </div>
      </CollapsableHeader>
      <Layout>
        <Codebook />
      </Layout>
    </Screen>
  );
};

CodebookScreen.propTypes = {
  layoutId: PropTypes.string,
  onComplete: PropTypes.func,
};

CodebookScreen.defaultProps = {
  layoutId: null,
  onComplete: () => {},
};

export default CodebookScreen;
