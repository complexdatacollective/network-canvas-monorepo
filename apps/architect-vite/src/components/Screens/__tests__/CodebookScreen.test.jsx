import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';

import React from 'react';
import { createStore } from 'redux';
import { render } from '@testing-library/react';
import CodebookScreen from '../CodebookScreen';

const mockState = {
  protocol: {
    present: {
      stages: [],
    },
  },
};

const mockProps = {
  store: createStore(() => mockState),
};

describe('<CodebookScreen />', () => {
  it('can render', () => {
    const { container } = render(<CodebookScreen {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
