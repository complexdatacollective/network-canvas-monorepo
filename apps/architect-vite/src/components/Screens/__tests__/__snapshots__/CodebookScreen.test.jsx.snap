// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CodebookScreen /> > can render 1`] = `
<div>
  <div
    class="screen-wrapper"
  >
    <div
      class="modal__background"
    />
    <div
      class="screen"
    >
      <div
        class="error"
      >
        <div
          class="error__layout"
        >
          <h1
            class="error__title"
          >
            Something went wrong.
          </h1>
          <div
            class="error__message"
          >
            <p>
              Could not find "store" in the context of "Connect(Codebook)". Either wrap the root component in a &lt;Provider&gt;, or pass a custom React context provider to &lt;Provider&gt; and the corresponding React context consumer to Connect(Codebook) in connect options.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
